### ISSUE TITLE
Using SMOTE before dividing into training and test sets

### ISSUE THREAD

**Issue by kkang2097** [georgymh/ml-fraud-detection#1]
If you do this, the oversampling algorithm will leak information from the test set onto the training sets, which explains why you got such a magical f1 score.

**Comment by georgymh** [comment]
You're totally right @kkang2097. I appreciate the correction. It's fixed now.
